bitrise/bitrise.go

```diff
@@ -50,23 +54,88 @@ type Environment struct {
 
 // App ...
 type App struct {
-	Slug, AccessToken string
+	BaseURL, Slug, AccessToken string
+	IsDebugRetryTimings        bool
+}
+
+// NewAppWithDefaultURL returns a Bitrise client with the default URl
+func NewAppWithDefaultURL(slug, accessToken string) App {
+	return App{
+		BaseURL:     "https://api.bitrise.io",
+		Slug:        slug,
+		AccessToken: accessToken,
+	}
+}
+
+// RetryLogAdaptor adapts the retryablehttp.Logger interface to the go-utils logger.
+type RetryLogAdaptor struct{}
+
+// Printf implements the retryablehttp.Logger interface
+func (*RetryLogAdaptor) Printf(fmtStr string, vars ...interface{}) {
```

Comment:
I think we could save some sort of string manipulation by using the LeveledLogger: https://github.com/hashicorp/go-retryablehttp/blob/master/client.go#L280

Wdyt?

Reply:
It does work, but not ideal, as that logger implements a kind of structured logger which does not have format strings.
`			switch v := logger.(type) {
			case Logger:
				v.Printf("[DEBUG] %s: retrying in %s (%d left)", desc, wait, remain)
			case LeveledLogger:
				v.Debug("retrying request", "request", desc, "timeout", wait, "remaining", remain)
			}`